src/internal/connector/graph/service.go

```diff
@@ -160,6 +172,40 @@ func HTTPClient(opts ...option) *http.Client {
 	return httpClient
 }
 
+// GetDefaultMiddlewares creates a new default set of middlewares for the Kiota request adapter
+func GetMiddlewares() []khttp.Middleware {
+	return []khttp.Middleware{
+		// khttp.NewRetryHandler(),
+		khttp.NewRedirectHandler(),
+		khttp.NewCompressionHandler(),
+		khttp.NewParametersNameDecodingHandler(),
+		khttp.NewUserAgentHandler(),
+		&LoggingMiddleware{},
+		&RetryHandler{
+			// add RetryHandlerOptions to provide custom operation like - MaxRetries and DelaySeconds
+			options: RetryHandlerOptions{ShouldRetry: func(delay time.Duration,
+				executionCount int, request *http.Request, response *http.Response,
+			) bool {
+				return true
+			}},
+		},
+	}
+}
+
+// GetDefaultMiddlewaresWithOptions creates a default slice of middleware for the Graph Client.
+func GetMiddlewaresWithOptions(options *msgraphgocore.GraphClientOptions) []khttp.Middleware {
+	kiotaMiddlewares := GetMiddlewares()
```

Comment:
nit: This should probably be `GetKiotaMiddlewares`, else it might be confused as a non "WithOptions" version of `GetMiddlewaresWithOptions`, or we could completely remove this indirection.